src/features/MentorPage/MentorPage.tsx

```diff
@@ -527,16 +538,42 @@ const mentorListCards: Array<ListCardProps> = mentorCards.map(item => {
   return cardProps;
 });
 
+export type handleSetVisibleCardProps = {
+  shouldShowMentorCard: boolean;
+  mentorCardData: ListCardProps | undefined;
+};
+
 const MentorPage = () => {
-  const [shouldShowMentorCard, setVisibleCard] = React.useState<{
-    [key: string]: unknown;
-  }>({});
+  const [shouldShowMentorCard, setVisibleCard] = React.useState(false);
+  const [currentCard, setCurrentCard] = React.useState<
+    ListCardProps | undefined
+  >();
 
-  const handleSetVisibleCard = (mentorCardData: ListCardProps) => {
-    setVisibleCard(mentorCardData);
-    console.log('shouldShowVisibleCard:', shouldShowMentorCard);
+  const handleSetVisibleCard = ({
+    shouldShowMentorCard,
+    mentorCardData,
+  }: handleSetVisibleCardProps) => {
+    setVisibleCard(shouldShowMentorCard);
+    setCurrentCard(mentorCardData);
   };
-
+  if (shouldShowMentorCard) {
+    return (
+      <PageLayout>
+        <MentorCard
+          setVisibleCard={handleSetVisibleCard}
+          mentorCardData={currentCard}
+        />
+        <OneContainerLayout headLine={mentorPageHeadline}>
+          <MentorInfoSearchDiv />
+          <MentorChips items={mentorChipList} />
+        </OneContainerLayout>
+        <MentorListItems
+          setVisibleCard={handleSetVisibleCard}
+          mentorCardData={mentorListCards}
+        />
+      </PageLayout>
+    );
+  }
```

Comment:
This is the parent for mentorCardData that is conditionally checked on each child. Just do it like this:

![image](https://user-images.githubusercontent.com/34128180/183650387-7c8d3996-fbcc-4cdc-8c70-11a560c2f33e.png)


See the currentCard type signature, it cannot be undefined.

Reply:
Also see that there is no need to have almost two identical renders behind a if-condition
```javascript
if(something) {
     return (
     <div>
         <this changes />
         <this is always here />
     </div>
     )
}
return (
    <div>
        <this is always here />
    </div>
    )
```

instead we can render the condition inline 
```javascript
return (
something ? <foo /> : <bar />
)
```

 OR with curly-brackets if middle of some content
 
 ```javascript
return ( <foo /> 
{somecondition && <hehee /> } 
<bar />
)
```